examples/multi_window_ref_app/lib/app/window_controller_render.dart

```diff
@@ -26,11 +30,33 @@ class WindowControllerRender extends StatelessWidget {
             key: key,
             onDestroyed: onDestroyed,
             onError: (String? reason) => onError(),
-            preferredSize: windowSettings.regularSize,
+            preferredSize: windowSettings.regularSizeNotifier.value,
             controller: controller as RegularWindowController,
             child: RegularWindowContent(
-                window: controller as RegularWindowController,
+                controller: controller as RegularWindowController,
                 windowSettings: windowSettings,
+                positionerSettingsModifier: positionerSettingsModifier,
+                windowManagerModel: windowManagerModel));
+      case WindowArchetype.popup:
+        return PopupWindow(
+            key: key,
+            onDestroyed: onDestroyed,
+            onError: (String? reason) => onError(),
+            preferredSize: windowSettings.popupSizeNotifier.value,
+            anchorRect: windowSettings.anchorToWindowNotifier.value
+                ? null
+                : windowSettings.anchorRectNotifier.value,
```

Comment:
If the anchor rect is not null, it must be clamped to the view size; otherwise the window positioning will not work as expected:
```suggestion
                : () {
                    final dpr = MediaQuery.of(context).devicePixelRatio;
                    return _clampRectToSize(
                        windowSettings.anchorRectNotifier.value,
                        windowManagerModel.selected!.view!.physicalSize / dpr);
                  }(),
```
where `_clampRectToSize` is
```dart
Rect _clampRectToSize(Rect anchorRect, Size? size) {
  double left = anchorRect.left.clamp(0, size?.width as double);
  double top = anchorRect.top.clamp(0, size?.height as double);
  double right = anchorRect.right.clamp(0, size?.width as double);
  double bottom = anchorRect.bottom.clamp(0, size?.height as double);
  return Rect.fromLTRB(left, top, right, bottom);
}
```
Although this suggestion works, it might be better to enforce the clamping within `PopupWindow` since it is a precondition for the positioning logic.

Reply:
What happens if we fail to clamp? Does it fail or does it just place incorrectly? That will change where I put this logic

Reply:
It won't fail. It'll just position it incorrectly depending on the anchoring used

Reply:
Ok. I think that is a good thing to put at the `PopupWindow` core widget point!

Reply:
Fixed 🎉